grin/src/seed.rs

```diff
@@ -286,29 +300,23 @@ fn connect_and_req(
 	h: reactor::Handle,
 	addr: SocketAddr,
 ) -> Box<Future<Item = (), Error = ()>> {
-	let connect_peer = p2p.connect_peer(addr, h).map_err(|_| ());
-	let timer = Timer::default();
-	let timeout = timer.timeout(connect_peer, Duration::from_secs(5));
+	let connect_peer = p2p.connect_peer(addr, h);
 	let p2p_server = p2p.clone();
-
-	let fut = timeout.then(move |p| {
+	let fut = connect_peer.then(move |p| {
 		match p {
 			Ok(Some(p)) => {
-				let p = p.read().unwrap();
-				let peer_result = p.send_peer_request(capab);
-				match peer_result {
-					Ok(()) => {}
-					Err(_) => {}
-				}
-			}
-			Err(_) => {
-				let update_result = p2p_server.update_state(addr, p2p::State::Defunct);
-				match update_result {
-					Ok(()) => {}
-					Err(_) => {}
+				debug!(LOGGER, "connect_and_req: ok, will attempt send_peer_request");
+				if let Ok(p) = p.try_read() {
+					let _ = p.send_peer_request(capab);
 				}
-			}
-			_ => {}
+			},
+			Ok(None) => {
+				error!(LOGGER, "connect_and_req: ok but none inner (what does this mean?), {}", addr);
+			},
+			Err(e) => {
+				error!(LOGGER, "connect_and_req: err - {:?}, {}, flagging as defunct", e, addr);
+				let _ = p2p_server.update_state(addr, p2p::State::Defunct);
```

Comment:
Logging as error will get people worried there's some real issue whereas this is bound to happen fairly often, no?

Reply:
:+1: will update to `debug` level.